### PR TITLE
Redo organelles with lower mesh count

### PR DESCRIPTION
**Pull request by HexapodPhilosopher**
**Brief Description of What This PR Does**

Adjusts models and shaders for organelles. Each organelle now uses a single model and texture. Several textures are now shared between models.

* Remade Nucleus - model adjustment and retexturing. Added pores and visible DNA. Made the Golgi body and ER stand out more. Now has a single model and texture. Poly count increased from 1438 to 2170.
* Redid Binding Agent with a single model and texture
* Remade Cilia as a single model, with fading at the base. Redid animation, and combined texture into Pilus texture.
* Adjusted Chemoplast. Fits better on the hex.
* Cleaned up Chemorecepter and added fading to base.
* Reduced Ferroplast from 3 models to 1. Removed Membrane texture.
* Added fading to Flagellum base. Made animation more curvy. Set to same texture as Pilus.
* Redid Lysosome as a single model.
* Remade Mitochondrion. Now more wiggly and more textured.
* Redid Nitroplast with half the polygons.
* Made shared texture for Nitoplast and Nitrogenase.
* Added fading to pilus and injectisome
* Redid Rusticyanin, sharing texture with Ferroplast.
* Reduced Thermoplast to single model and texture. Particles made more visible, and emission added.
* Redid Thylakoids and Chloroplast using a single texture. Chloroplast texturing is now more consistent, and poly count reduced by 40%. Thylakoid poly count reduced by 20%.
* Changed Toxisome to use same texture as Toxin Vacuole.

* Re-oriented all organelles to match the shader animation.

	
Shader improvements:
* Changed the angle of the shader vertex animation, so that default behavior (jiggling about in all directions when seen from above) works without the organelles having to be rotated first. This will make it easier to add new parts.
* Added scaling to dissolve effect, causing smaller or larger pieces of the mesh to be visible while dissolving.
* Clamped min and max values for emission, to remove bright spots that sometimes appear due to the dissolve shader
* Redid Fresnel effect on organelles, to give more control over visible insides. Now 0 = no effect, 1 = full effect.
* Added emissive organelle shader, used for thermoplast particle glow.

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [X] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: assets/models/organelles/Cytoplasm.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Cytoplasm.tscn:15]: Is it intentional that this model has fresnel mode active? If I remember right that didn't used to be enabled for anything in the game.

File: assets/textures/ExternalOrganelle.png
Comment (hhyyrylainen) [inline assets/textures/ExternalOrganelle.png:1]: Why is this named external organelle? Wasn't `Pilus.png` a good name?

File: assets/textures/Nucleus2.png.import
Comment (hhyyrylainen) [inline assets/textures/Nucleus2.png.import:1]: What's the difference between this and the other nucleus texture? If both are needed could the names be clarified a bit?

File: assets/textures/PhotosynthesisOrganelle.png
Comment (hhyyrylainen) [inline assets/textures/PhotosynthesisOrganelle.png:1]: What's the reasoning with these very generic names? Usually it makes it much more clear to name the files after the organelle they are used in.

File: shaders/OrganelleOpaque.gdshader
Comment (hhyyrylainen) [inline shaders/OrganelleOpaque.gdshader:8]: I assume the scale is 0-1 here? If so could you add a `hint_range` for this? That makes the value show up more intuitively in the Godot editor.

File: shaders/OrganelleTransparent.gdshader
Comment (hhyyrylainen) [inline shaders/OrganelleTransparent.gdshader:8]: I think these other shader files also have the same thing about the hint range.

File: shaders/OrganelleEmissive.gdshader
Comment (hhyyrylainen) [inline shaders/OrganelleEmissive.gdshader:49]: Variables should be named in camelCase and not snake_case

Comment (hhyyrylainen) [reply shaders/OrganelleEmissive.gdshader:49]: Though is there a reason why this needs to be a variable? It looks like the UV is not modified so why not just use `UV` in the places this is used?

File: shaders/OrganelleEmissive.gdshader
Comment (hhyyrylainen) [inline shaders/OrganelleEmissive.gdshader:64]: There's multiple snake_case variable names here.

File: assets/models/organelles/Flagellum.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Flagellum.tscn:44]: Why is the default speed scale gone? The intention with that is that the flagellum only plays the full speed swimming animation when the C# code commands it, and otherwise is like in the "idle" state, as there's many more cases where it is useful to have the idle animation than needing to show the max speed.

File: assets/models/organelles/Nucleus.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Nucleus.tscn:10]: Is this required? This may cause unintended rendering on top of organelles that are in front o the nucleus. The reason this was used before is that the nucleus was multiple parts and it was required to get the draw order for them correct.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Overall this looks to be in pretty good shape already, but I had a few correction suggestions. Though I didn't test in game yet to see if there's some unexpected issues I couldn't see from just the text data.

Also did you test in opengl3 mode? That needs to at least look decent after these major graphics updates. The one new shader is the most likely thing that *could* have issues there.

File: assets/models/organelles/Cytoplasm.tscn
Comment (HexapodPhilosopher) [reply assets/models/organelles/Cytoplasm.tscn:15]: That's just the parameter that's been in OrganelleOpaque.gdshader since last year at least. Some of the membranes (including the Ferroplast one) were using it previously.
I put it here just to make it more transparent, but I could just make the texture more transparent instead.

File: assets/textures/ExternalOrganelle.png
Comment (HexapodPhilosopher) [reply assets/textures/ExternalOrganelle.png:1]: I changed it because both forms of pilus as well as the flagellum and cilia now all use that texture.

File: assets/textures/Nucleus2.png.import
Comment (HexapodPhilosopher) [reply assets/textures/Nucleus2.png.import:1]: Damn, that should have been removed.

File: shaders/OrganelleOpaque.gdshader
Comment (HexapodPhilosopher) [reply shaders/OrganelleOpaque.gdshader:8]: 1.0 is the default, but values above 1 decrease the size of the noise. Putting it like that, maybe I should have called it 'dissolveResolution'...
Values from 0.5 to 5.0 are fine.

File: shaders/OrganelleEmissive.gdshader
Comment (HexapodPhilosopher) [reply shaders/OrganelleEmissive.gdshader:49]: I saw that used somewhere (outside Thrive) and thought it might reduce the number of times it has to check the mesh. I didn't find info on that.

File: assets/models/organelles/Flagellum.tscn
Comment (HexapodPhilosopher) [reply assets/models/organelles/Flagellum.tscn:44]: That was unintended. It seemed to work without it, though.

**Comment by HexapodPhilosopher** [comment]
I didn't test it in OpenGL3 mode. I'll have to do that now.

File: assets/models/organelles/Nucleus.tscn
Comment (HexapodPhilosopher) [reply assets/models/organelles/Nucleus.tscn:10]: I was using it while trying out different ways to do the nucleus. It's unnecessary now.

File: assets/models/organelles/Cytoplasm.tscn
Comment (hhyyrylainen) [reply assets/models/organelles/Cytoplasm.tscn:15]: Yeah I know the parameter was added ages ago but it was removed from all of the organelles. I suppose if it looks fine and the dissolve effect also looks fine with it on for specific organelles, it can be kept on.

File: assets/textures/ExternalOrganelle.png
Comment (hhyyrylainen) [reply assets/textures/ExternalOrganelle.png:1]: The slime jet is also an external organelle, so I still think this should be have a better name as that doesn't use it. Maybe like `StrandOrganelle.png` as that would slightly narrow it down?

File: assets/textures/PhotosynthesisOrganelle.png
Comment (hhyyrylainen) [reply assets/textures/PhotosynthesisOrganelle.png:1]: That sounds like good justification, it just isn't immediately clear to me what it is for. But now that I know, I approve of the new name.

File: shaders/OrganelleOpaque.gdshader
Comment (hhyyrylainen) [reply shaders/OrganelleOpaque.gdshader:8]: I think the name is clear enough, but you should add `hint_range(0.5, 5)` here to make the Godot editor show a slider for picking the value.

File: shaders/OrganelleEmissive.gdshader
Comment (hhyyrylainen) [reply shaders/OrganelleEmissive.gdshader:49]: I think the way it works is that godot just setups the variable once before starting the fragment shader. So I highly think it doesn't matter. And if I remember right in like GLSL or HLSL you need to specify the UV as a method parameter so it is pretty clear there that the data is copied from the mesh into local memory for the shader when setting it up so again I highly doubt it affects used memory bandwidth.

File: assets/models/organelles/Flagellum.tscn
Comment (hhyyrylainen) [reply assets/models/organelles/Flagellum.tscn:44]: I'm 90% sure that the editor would end up displaying the animation playing at 100% speed which looks very busy and not very passive in the editor. So this change would increase visual business in the editor (if I'm right in how I expect this to affect things).

File: assets/models/organelles/Nucleus.tscn
Comment (hhyyrylainen) [reply assets/models/organelles/Nucleus.tscn:10]: Alright so it can be removed, I'll wait for your commit changing that.

File: assets/models/organelles/Cytoplasm.tscn
Comment (HexapodPhilosopher) [reply assets/models/organelles/Cytoplasm.tscn]: Was there a specific reason it was removed before?
Also, I forgot to mention that I altered the calculation for it slightly.
It's now used only by lysosome, Signalling Agent, Vacuole and the endosome.

Comment (hhyyrylainen) [reply assets/models/organelles/Cytoplasm.tscn]: I can't remember / know, but I think it was removed as it looked bad. So it's fine to put it back in if it looks good now.

**Comment by HexapodPhilosopher** [comment]
The Thermoplast issue seems to be because of the lighting issue on my system. The emission is working outside the editor, and is inactive in the editor.
I think I'll stick with it, and I'll just have to get feedback on if anyone else has issues.

File: assets/models/organelles/Cilia.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Cilia.tscn:102]: This also has the changed speed scale like flagellum. I think you fixed that there, but this is still like this...

File: assets/models/organelles/Lysosome.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Lysosome.tscn:10]: Another render priority I spotted...

File: assets/models/organelles/Nucleus.tscn
Comment (hhyyrylainen) [inline assets/models/organelles/Nucleus.tscn:27]: Skeleton? Why does the Nucleus have a skeleton set.

**Comment by hhyyrylainen** [comment]
What's the problem with the thermoplast? Does the new emission shader not work or? I noticed just a few more small technical details but all of the other files I didn't comment on now seem fine to me so this is very close to being done. BTW if you have the models as .blend files for example (or another modelling program format) could you upload those files to https://dev.revolutionarygamesstudio.com/files/Raw%20Assets/Models/Organelles so that in the future there's the raw source files if any future tweaks are required for the organelle models?

File: assets/models/organelles/Nucleus.tscn
Comment (HexapodPhilosopher) [reply assets/models/organelles/Nucleus.tscn:27]: It doesn't have a skeleton. I'm not sure how that got added.

**Comment by HexapodPhilosopher** [comment]
The emission shader works fine, except in the cell editor when in OpenGL mode (where the emission seems to be inactive).
After the Bloom slider PR was added, the maximum effect of the emission in terms of the glow around the emitting object seems much lower. That doesn't make a noticeable difference for the Thermoplast, though.

**Comment by hhyyrylainen** [comment]
This is probably super hard to find but Godot doesn't fully like one of the new models I think as I got this when first importing all the new changes: `scene/resources/3d/importer_mesh.cpp:521 - Ignoring face with non-finite normal in LOD generation.`

(I'm now doing some final checks on this before merging)

**Review by hhyyrylainen — APPROVED** [review]
I tested in opengl mode and only really the thermoplast looked bad, kind of like a disgusting biological thing actually but I think that's good enough for not officially supported mode (which is still totally playable). I looked at all the models in game and I think they work fine. I noticed a slight edge in the slime jet model at some very specific angles in the game, but other than that I don't have anything really comment on the models so good job! 

I'll merge this now but if at all possible I'd still like to see the source 3D models added to the devcenter so that if someone else in like 5 years needs to modify an organelle slightly that would be possible to do (https://dev.revolutionarygamesstudio.com/files/Raw%20Assets/Models/Organelles).

**Pull request merged**
